tools/autograd/derivatives.yaml

```diff
@@ -1065,10 +1065,15 @@
 
 - name: masked_scatter(Tensor self, Tensor mask, Tensor source) -> Tensor
   self: grad.masked_fill(mask, 0)
-  source: masked_scatter_backward(grad, mask, source.sym_sizes())
+  source: masked_scatter_backward_symint(grad, mask, source.sym_sizes())
   mask: non_differentiable
   result: self_t.masked_scatter(mask, source_t)
 
+- name: masked_scatter_backward(Tensor grad_output, Tensor mask, SymInt[] sizes) -> Tensor
+  grad_output: zeros_like(grad_output).masked_scatter(mask, grad)
```

Comment:
```suggestion
  grad_output: zeros_like(grad_output).masked_scatter_(mask, grad)
```

Reply:
This causes errors: 

```
While computing batched gradients, got: vmap: aten::masked_scatter_(self, *extra_args) is not possible because there exists a Tensor `other` in extra_args that has more elements than `self`. This happened due to `other` being vmapped over but `self` not being vmapped over at level 1. Please try to use out-of-place operators instead of aten::masked_scatter_.
```